sdk/eventhub/event-hubs/src/eventHubBufferedProducerClient.ts

```diff
@@ -0,0 +1,315 @@
+   */
+  constructor(
+    fullyQualifiedNamespace: string,
+    eventHubName: string,
+    credential: TokenCredential | NamedKeyCredential | SASCredential,
+    options: EventHubBufferedProducerClientOptions // eslint-disable-line @azure/azure-sdk/ts-naming-options
+  );
+  constructor(
+    fullyQualifiedNamespaceOrConnectionString1: string,
+    eventHubNameOrOptions2: string | EventHubBufferedProducerClientOptions,
+    credentialOrOptions3?:
+      | TokenCredential
+      | NamedKeyCredential
+      | SASCredential
+      | EventHubBufferedProducerClientOptions,
+    options4?: EventHubBufferedProducerClientOptions // eslint-disable-line @azure/azure-sdk/ts-naming-options
+  ) {
+    this._context = createConnectionContext(
+      fullyQualifiedNamespaceOrConnectionString1,
+      eventHubNameOrOptions2,
+      credentialOrOptions3,
+      options4
+    );
+    if (typeof eventHubNameOrOptions2 !== "string") {
+      this._clientOptions = eventHubNameOrOptions2 || {};
+    } else if (!isCredential(credentialOrOptions3)) {
+      this._clientOptions = credentialOrOptions3 || {};
+    } else {
+      this._clientOptions = options4 || {};
+    }
+  }
+
+  /**
+   * Closes the AMQP connection to the Event Hub instance,
+   * returning a promise that will be resolved when disconnection is completed.
+   *
+   * @param options - The set of options to apply to the operation call.
+   * @returns Promise<void>
+   * @throws Error if the underlying connection encounters an error while closing.
+   */
+  async close(options: BufferedCloseOptions = {}): Promise<void> {
+    if (options.flush) {
+      await this.flush(options);
+    }
+    return this._context.close();
+  }
+
+  /**
+   * Enqueues events into the buffer to be published to the Event Hub.
+   * If there is no capacity in the buffer when this method is invoked,
+   * it will wait for space to become available and ensure that the events
+   * have been enqueued.
+   *
+   * When this call returns, the events have been accepted into the buffer,
+   * but it may not have been published yet.
+   * Publishing will take place at a nondeterministic point in the future as the buffer is processed.
+   *
+   * @param events - An array of {@link EventData} or `AmqpAnnotatedMessage`.
+   * @param options - A set of options that can be specified to influence the way in which
+   * events are sent to the associated Event Hub.
+   * - `abortSignal`  : A signal the request to cancel the send operation.
+   * - `partitionId`  : The partition this set of events will be sent to. If set, `partitionKey` can not be set.
+   * - `partitionKey` : A value that is hashed to produce a partition assignment. If set, `partitionId` can not be set.
+   * @returns The total number of events that are currently buffered and waiting to be published, across all partitions.
+   */
+  async enqueueEvents(
+    events: EventData[] | AmqpAnnotatedMessage[],
```

Comment:
No single message at a time option?

Reply:
There should be;  we  want to encourage the single overload as the majority case for callers.  The multiple overload has the possibility for partial success; we want to guide callers to the singular overload  for scenarios where tracking the success/failure of each event being enqueued is important to the application.

Reply:
I'll add the singular `enqueueEvent` method.

Reply:
@chradek if other languages support single message option as an overload, is it worth it to support it here as such as well?
```ts
enqueueEvents(events: EventData | AmqpAnnotatedMessage, options: EnqueueEventOptions): Promise<number>;
```

Reply:
Sure, I'll add that in for the preview. For what it's worth, we debated this when looking at the traditional API send API. We got rid of the single event overload there since it seemed easy enough to just wrap an event in an array, but happy to try it out again now that the use case is changed.

Reply:
Indeed we did.  However, this use case is decidedly different and the single event enqueue is the preferred call pattern for most customer scenarios.   😄